### ISSUE TITLE
ArrayObject is restricted (Magento Code Scans)

### ISSUE THREAD

**Issue by florisschreuder** [Smile-SA/elasticsuite#2550]
Using Magento 2 code scans on our own modules showed that the code scan fails with this note:

```
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 72 | ERROR | Class 'ArrayObject' is restricted in
    |       | /opt/atlassian/pipelines/agent/build/Model/Rule/Indexer/Fulltext.php.
    |       | Suggested replacement: Custom class, extended from
    |       | ArrayObject with overwritten serialize/unserialize
    |       | methods
----------------------------------------------------------------------
```

I've not managed to find a direct source, but the Upgrade Compatibility Tool does fail/note the same thing:
https://experienceleague.adobe.com/docs/commerce-operations/upgrade-guide/upgrade-compatibility-tool/error-messages.html?mt=false

Our code in this case was based on the Elasticsuite code, both in the core (below) as well as in the CMS Pages module.
https://github.com/Smile-SA/elasticsuite/blob/e8e67b407f3f49a2bac16e3525173d19af9c0f7c/src/module-elasticsuite-catalog/Model/Category/Indexer/Fulltext.php#L86

Adobe suggests creating a custom class which extends ArrayObject, which seems a fairly easy thing to do. But it would be prettiest if it happens once within core Elasticsuite I think.

### Preconditions

Magento Version : 2.4.x

ElasticSuite Version : 2.10.x

Environment : not applicable

Third party modules : not applicable

### Steps to reproduce

Run Magento's PHPCS standard=magento2 scan.

### Expected result

1. Success

### Actual result

1. Failure, with comment named above

**Comment by romainruaud** [comment]
@vahonc you can create an object inheriting the legacy ArrayObject

**Comment by florisschreuder** [comment]
Hi @vahonc ,

At Experius we use Bitbucket Pipelines on all modules we create ourselves. These pipelines run PHP CS tests, including the ones based on Magento's Coding Standard.

https://developer.adobe.com/commerce/marketplace/guides/sellers/code-sniffer/
https://github.com/magento/magento-coding-standard

This particular error we got on a module we created to index a new entity (CMS pages for our own content module). We based that module on your CMS-page-indexer module, and because of this we only got the particular ArrayObject issue.

We've currently got those lines mentioned with 'PHPCS Ignore', but that'll cause problems in certain cases where we want Adobe/Magento to actually run codescans.

Thanks!
